src/frames.c

```diff
@@ -307,8 +307,9 @@ static int is_frame_initialized(const novas_frame *frame) {
 int novas_make_frame(enum novas_accuracy accuracy, const observer *obs, const novas_timespec *time, double dx, double dy,
         novas_frame *frame) {
   static const char *fn = "novas_create_frame";
-  static const object earth = { NOVAS_PLANET, NOVAS_EARTH, "Earth" };
-  static const object sun = { NOVAS_PLANET, NOVAS_SUN, "Sun" };
+  static const cat_entry zero_star = {0};
+  static const object earth = { NOVAS_PLANET, NOVAS_EARTH, "Earth", zero_star };
```

Comment:
Is it in response to a compiler warning? Otherwise, I'm not sure this is necessary. You are correct that the initializer needs to cover at least one element (although gcc seems to not require even that), but after the defined elements, all the remaining footprint of the structure is initialized to 0, as if by `memset()`. So, there should be no need to explicitly define null substructures for the initializer. I think it's just unnecessarily verbose, but I might be wrong...

Reply:
This one is tricky - I try to lean heavy into the "explicit is better than implicit camp", and -Wextra does as well - throwing a warning on partial initialization. You'd think we could just slap a `{0}` at the end and call it a day, but apparently there is an old GCC bug about that? https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119

But I agree it is a little on the extra verbose side - although I've been mulling about the design of `object` a bit still. It would be better if it weren't possible to construct an invalid `object` at all - meaning a sidereal object must contain a valid `cat_entry` while ephem stuff must not contain it, at a type level. I don't think that's possible in C, though.

Reply:
This is all of course my opinion, and you are welcome to cherry-pick any of this stuff haha

Reply:
Yes, if -`Wextra` complains, then be it. I did have `-Wextra` among the compiler flags originally, until I realized that older gccs did not have the flag, and gave an error as a result. So, I removed it. Maybe, I can reinstate it for the CI build though...